guard/src/rules/eval_context_tests.rs

```diff
@@ -464,3 +452,197 @@ fn test_with_converter() -> Result<()> {
 
     Ok(())
 }
+
+#[test]
+fn test_handle_function_call() -> Result<()> {
+    let path_value = PathAwareValue::try_from(serde_yaml::from_str::<serde_yaml::Value>(
```

Comment:
It's okay for this PR, but in future we might want to modularize this. Add a `@TO-DO` comment?